### PR TITLE
UI fixes

### PR DESCRIPTION
**Pull request by stewsk**
Address some UI related issues:

* add IDs for Header Facets
* replace outdated manifest parameter
* replace deprecated annotations 
* remove unused anotations
* add compact filter definition
* consistently use localized labels
* add value help for visual filters/compact filters
* update minimum UI5 version

### CODE REVIEW

**Comment by stewsk** [comment]
Open questions:
* how can I rename the (default?) measure "sum prices"?
* how do I get reasonable value helps for visual filters status and airline?

**Comment by MN-public** [comment]
> how can I rename the (default?) measure "sum prices"?

Could you please explain what you mean? The property name or the label seen in runtime?

> how do I get reasonable value helps for visual filters status and airline?


**Value Help for Status**:
1. Expose BookingStatus as a projection in analytics-service.cds
` entity BookingStatus as projection on my.BookingStatus`

2. You can use the Page Editor to add a value help for the compact filter Booking Status in the Display Type property. 
image
image

The resulting annotations look as follows:

```
annotate service.Bookings with {
    status @(Common.ValueList : {
            $Type : 'Common.ValueListType',
            CollectionPath : 'BookingStatus',
            Parameters : [
                {
                    $Type : 'Common.ValueListParameterInOut',
                    LocalDataProperty : status,
                    ValueListProperty : 'code',
                },
            ],
        },
        Common.ValueListWithFixedValues : false
)};
```
**Value Help for Airline:**
Same as p2 above. You can add additional information to the results list. 
image
The resulting annotations:
```
annotate service.Bookings with {
    airline @(Common.ValueList : {
            $Type : 'Common.ValueListType',
            CollectionPath : 'Airline',
            Parameters : [
                {
                    $Type : 'Common.ValueListParameterInOut',
                    LocalDataProperty : airline,
                    ValueListProperty : 'AirlineID',
                },
                {
                    $Type : 'Common.ValueListParameterDisplayOnly',
                    ValueListProperty : 'Name',
                },
                {
                    $Type : 'Common.ValueListParameterDisplayOnly',
                    ValueListProperty : 'CurrencyCode_code',
                },
            ],
        },
        Common.ValueListWithFixedValues : false
)};
```

**Comment by stewsk** [comment]
>>how can I rename the (default?) measure "sum prices"?

>Could you please explain what you mean? The property name or the label seen in runtime?

In the settings for the chart, for example: there is a default measure "Flight Price" which I'd prefer to be called "Total price"
-> is there a way to change that name?

image

**Comment by MN-public** [comment]
@stewsk 
> In the settings for the chart, for example: there is a default measure "Flight Price" which I'd prefer to be called "Total price" -> is there a way to change that name?
> 

If the measure label is defined on the same app, you can change it directly in the Properties pane of the Page editor. Just enter a new text in the label field and put the cursor out. Then you can click on the Globe button to prepare it for translation:
image


If the measure label comes from the different app, you cannot change it from the Page Editor UI for the current app to avoid cross-app dependencies. You can change it from the app it is defined in or just navigate to the source code and change it there manually. 
image
Please keep in mind that if it is a custom measure based on the property, the changed label will apply to all occurrences of the field unless a different label is applied for that occurrence (e.g. column name). 
image

I believe it makes sense to use the transformation measures for the standard aggregation methods like sum:
```
  Analytics.AggregatedProperty #sumPrice :
   {
    Name                 : 'sumPrice',
    AggregationMethod    : 'sum',
    AggregatableProperty : price,
    @Common.Label        : 'test'
  }
```
 In that case you will not have such an unwanted side effect. But I'm not sure if you still need a custom aggregate for the KPI

File: app/travel_analytics/annotations.cds [hunk 1]
```diff
@@ -381,3 +378,10 @@ annotate service.Bookings with @UI : {
     { Value : Distance,       },
   ]},
 };
+
+// determines the order of visual filters
+annotate service.Bookings with @UI.SelectionFields : [
+  FlightDate,
+  status,
+  airline
+];
```
[the remark this answers is not in the document]
Comment (stewsk) [reply app/travel_analytics/annotations.cds:394]: Hi @MN-public , please see https://github.com/SAP-samples/cap-sflight/pull/494

**Pull request merged**